### DISCUSSION TITLE
Resource for Financial Institutions

### DISCUSSION THREAD

**Discussion by danpastori** [serversideup/financial-freedom#114] [Q&A]
Hi!

Just curious if someone has a resource we could use to seed the Financial Freedom database with financial institutions?

To get started I'm going to kick off the install with around 20-30 of the top institutions in the US with the ability to quickly add more. This will suffice for now. 

In the future I'd like the ability to seed the database based off of country or location. The goal is to keep as much data hosted locally so it can easily be run anywhere and keep you in control of the data. I'd be open to any thoughts others have as well.

Thanks!

**Comment by PrplHaz4** [comment]
> Hi!
> 
> Just curious if someone has a resource we could use to seed the Financial Freedom database with financial institutions?
> 
> To get started I'm going to kick off the install with around 20-30 of the top institutions in the US with the ability to quickly add more. This will suffice for now.
> 
> In the future I'd like the ability to seed the database based off of country or location. The goal is to keep as much data hosted locally so it can easily be run anywhere and keep you in control of the data. I'd be open to any thoughts others have as well.
> 
> Thanks!

Fed Listing: https://www.federalreserve.gov/releases/lbr/current/default.htm
OFX Info: https://www.ofxhome.com/index.php/home/directory

```
Bank Name / Holding Co Name 	Nat'l Rank 	Bank ID 	Bank Location 	Charter 	Consol Assets (Mil $) 	Domestic Assets (Mil $) 	Pct Domestic Assets 	Pct Cumulative Assets 	Domestic Branches 	Foreign Branches 	IBF
JPMORGAN CHASE BK NA/JPMORGAN CHASE & CO 	1 	852218 	COLUMBUS, OH 	NAT 	3,385,581 	2,613,981 	77 	16 	4,871 	32 	Y
BANK OF AMER NA/BANK OF AMER CORP 	2 	480228 	CHARLOTTE, NC 	NAT 	2,465,234 	2,336,157 	95 	27 	3,785 	23 	Y
WELLS FARGO BK NA/WELLS FARGO & CO 	3 	451965 	SIOUX FALLS, SD 	NAT 	1,704,891 	1,678,100 	98 	35 	4,430 	10 	Y
CITIBANK NA/CITIGROUP 	4 	476810 	SIOUX FALLS, SD 	NAT 	1,657,372 	1,014,005 	61 	43 	659 	121 	Y
U S BK NA/U S BC 	5 	504713 	CINCINNATI, OH 	NAT 	657,184 	646,725 	98 	46 	2,317 	1 	N
PNC BK NA/PNC FNCL SVC GROUP 	6 	817824 	WILMINGTON, DE 	NAT 	553,114 	551,126 	100 	48 	2,431 	1 	N
GOLDMAN SACHS BK USA/GOLDMAN SACHS GROUP THE 	7 	2182786 	NEW YORK, NY 	SMB 	538,127 	463,989 	86 	51 	2 	2 	N
TRUIST BK/TRUIST FC 	8 	852320 	CHARLOTTE, NC 	SNM 	535,000 	534,933 	100 	53 	2,001 	0 	N
CAPITAL ONE NA/CAPITAL ONE FC 	9 	112837 	MC LEAN, VA 	NAT 	468,780 	468,225 	100 	55 	283 	1 	N
T D BK NA/TD GRP US HOLDS LLC 	10 	497404 	WILMINGTON, DE 	NAT 	366,252 	366,252 	100 	57 	1,175 	0 	N
```

**Reply by danpastori** [reply]
Thanks so much! This looks like a great resource!

**Reply by PrplHaz4** [reply]
If you aren't familiar already it may be worth looking a the GNUCash OFX Import setup - it seems to be by far the most mature for bank transaction imports - but unfortunately isn't very portable/modular/modern-web-compatible.

For additional OFX/bank import the OFXTools discussions are a treasure trove of info - mostly arriving at it being really hard to keep up with automated banking imports. https://github.com/csingley/ofxtools/discussions

It seems like many people are now resorting to roll their own tools to pull things down via browser automation :-/

**Reply by danpastori** [reply]
I'm totally not familiar with OFX and GNU Cash. Most of the tools I've seen have been very legacy like a lot of banking software. I'll have to check out some of their info though, this looks incredible! 

I don't mind contributing or modifying to fit the needs of Financial Freedom to decentralize and modernize.

**Comment by RokSiEu** [comment]
This will end up in huge list which will be hard to maintain to be up to date. 

For EU this one can be helpful, it has CSV files per country, but like said before, it's already outdated, I've checked wikipedia and few other resources, including PDF file found on ECB site, also outdated 🤔 

https://www.bde.es/wbe/en/estadisticas/otras-clasificaciones/clasificacion-entidades/listas-instituciones-financieras/listas-instituciones-financieras-monetarias-pais/

**Reply by danpastori** [reply]
For sure. Banks change all the time. This looks like an excellent resource for EU! I'm just trying to seed with some of the bigger ones so there's there's a launching point. Adding an institution will be super easy within Financial Freedom
